src/NUnitFramework/framework/Internal/Tests/TestSuite.cs

```diff
@@ -100,7 +100,15 @@ public void Sort()
         {
             if (!MaintainTestOrder)
             {
-                this.tests.Sort();
+                this.tests.Sort(delegate (ITest x, ITest y)
+                {
+                    var xKey = 0;
+                    var yKey = 0;
+                    if (x.Properties.ContainsKey("order")) int.TryParse(x.Properties["order"][0].ToString(), out xKey);
+                    if (y.Properties.ContainsKey("order")) int.TryParse(y.Properties["order"][0].ToString(), out yKey);
+                    return xKey.CompareTo(yKey) != 0
+                    ? xKey.CompareTo(yKey) : x.Name.CompareTo(y.Name); ;
```

Comment:
I'm not sure about retaining the name sort, but see the general comment before making more changes.